be/src/exec/tablet_sink_index_channel.cpp

```diff
@@ -493,16 +509,19 @@ Status NodeChannel::add_chunks(Chunk* input, const std::vector<std::vector<int64
     }
 
     SCOPED_TIMER(_ts_profile->pack_chunk_timer);
+
     // 1. append data
-    _cur_chunk->append_selective(*input, indexes.data(), from, size);
+    _append_data_to_cur_chunk(*input, indexes.data(), from, size);
+
     for (size_t index_i = 0; index_i < index_tablet_ids.size(); ++index_i) {
         auto req = _rpc_request.mutable_requests(index_i);
         for (size_t i = from; i < size; ++i) {
             req->add_tablet_ids(index_tablet_ids[index_i][indexes[from + i]]);
         }
     }
 
-    if (_cur_chunk->num_rows() < _runtime_state->chunk_size()) {
+    if (_cur_chunk->num_rows() < _runtime_state->chunk_size() &&
+        _cur_chunk_mem_usage < config::max_tablet_write_chunk_bytes) {
```

Comment:
Why not just use `_cur_chunk->byte_size() < config::max_tablet_write_chunk_bytes`?

Reply:
> byte_size

invoking `byte_size` may have additional overhead, so I choose to maintain this info during update